Sources/OpenAI/Private/StreamingSession.swift

```diff
@@ -46,40 +48,75 @@ final class StreamingSession<ResultType: Codable>: NSObject, Identifiable, URLSe
         guard let stringContent = String(data: data, encoding: .utf8) else {
             onProcessingError?(self, StreamingError.unknownContent)
             return
-        }
-        let jsonObjects = stringContent
+        }                
+        
+        // As data comes in, we process it by trying to decode to JSON. Since we may not have yet received a fully valid
+        // JSON string yet, we keep appending content if we're unable to decode due to a "dataCorrupted" error, trying again
+        // on the next pass in case that new data has completed the string to create valid (parsable) JSON, and so on.
+        // Note that while OpenAI's API doesn't appear to return partial string fragments, proxy service such as Helicone
+        // do seem to do so, making this logic necessary.
+        accumulatedData.append(stringContent)
```

Comment:
Could you extract this change to a separate PR, please?
Looks like this is a separate fix, that doesn't apply to custom headers.